src/unix/fs.c

```diff
@@ -393,6 +396,28 @@ static ssize_t uv__fs_readlink(uv_fs_t* req) {
   char* buf;
   char* newbuf;
 
+#if defined(UV__FS_PATH_MAX_FALLBACK)
+  /* We may not have a real PATH_MAX.  Read size of link.  */
+  struct stat st;
+  int ret;
+  ret = lstat(req->path, &st);
+  if (ret != 0)
+    return -1;
+  if (!S_ISLNK(st.st_mode)) {
+    errno = EINVAL;
+    return -1;
+  }
+
+  maxlen = st.st_size;
+
+  /* According to readlink(2) lstat can report st_size == 0
+     for some symlinks, such as those in /proc or /sys.  */
+  if (maxlen == 0)
+    maxlen = uv__fs_pathmax_size(req->path);
+#else
+  maxlen = uv__fs_pathmax_size(req->path);
+#endif
+
   maxlen = uv__fs_pathmax_size(req->path);
```

Comment:
This line is not needed anymore, right?

Reply:
Thanks.  That was a leftover from rebase conflicts.  Fixed.